### PR TITLE
feat/SOF-6631: prepare libs for hp.x

### PR DESCRIPTION
**Pull request by pranabdas**

### CODE REVIEW

File: src/application.js [hunk 1]
```diff
@@ -96,9 +96,21 @@ export class Application extends NamedDefaultableInMemoryEntity {
...
+        const allowedExecutables = executableList.filter((exe) => {
+            const { supportedApplicationVersions } = exe._json;
+            return (
+                !supportedApplicationVersions ||
+                (supportedApplicationVersions &&
+                    supportedApplicationVersions.includes(this.prop("version")))
+            );
+        });
+        return allowedExecutables;
+        // return Object.keys(tree).map((key) => {
```
Comment (timurbazhirov) [inline src/application.js:111]: Remove pls

File: src/application.js [hunk 2]
```diff
@@ -96,9 +96,21 @@ export class Application extends NamedDefaultableInMemoryEntity {
 
     get executables() {
         const tree = getAppTree(this.prop("name"));
-        return Object.keys(tree).map((key) => {
+        const executableList = Object.keys(tree).map((key) => {
             return new this.constructor.Executable({ ...tree[key], name: key });
         });
+        const allowedExecutables = executableList.filter((exe) => {
+            const { supportedApplicationVersions } = exe._json;
```
Comment (timurbazhirov) [inline src/application.js:103]: Let's use `.toJSON()` instead pls, `_json` is meant as an internal class var

File: src/application.js [hunk 3]
```diff
@@ -96,9 +96,18 @@ export class Application extends NamedDefaultableInMemoryEntity {
 
     get executables() {
         const tree = getAppTree(this.prop("name"));
-        return Object.keys(tree).map((key) => {
+        const executableList = Object.keys(tree).map((key) => {
```
Comment (spectre007) [inline src/application.js:99]: No need to add extra variables; we can do it in one go. Instantiating and then filtering as well as calling `toJSON()` will `deepclone` the executable config unnecessarily; I think we could do sth like this to minimize that:
```javascript
return Object.keys(tree)
    .filter(key) => {
        const { supportedApplicationVersions } = tree[key];
        return (
            !supportedApplicationVersions ||
            (supportedApplicationVersions &&
                supportedApplicationVersions.includes(this.prop("version")))
        );
    })
    .map((key) => {
            return new this.constructor.Executable({ ...tree[key], name: key });
    });
```

File: src/executable.js [hunk 4]
```diff
@@ -41,4 +41,16 @@ export class Executable extends mix(NamedDefaultableInMemoryEntity).with(Runtime
     getFlavorByConfig(config) {
         return config ? this.getFlavorByName(config.name) : this.defaultFlavor;
     }
+
+    getFlavorByApplicationVersion(version) {
+        const filteredFlavors = this.flavors.filter((flavor) => {
+            const { supportedApplicationVersions } = flavor.toJSON();
```
Comment (spectre007) [inline src/executable.js:47]: Let's avoid calling `toJSON()` if not absolutely necessary. We can use
```javascript
const supportedApplicationVersions = flavor.prop("supportedApplicationVersions");
```
instead.

**Review by spectre007 — COMMENTED** [review]
minor changes suggested, looks good otherwise.
Note: the version of `application-flavors` needs to be replaced with the published version before merging.

File: src/executable.js [hunk 5]
```diff
@@ -41,4 +41,16 @@ export class Executable extends mix(NamedDefaultableInMemoryEntity).with(Runtime
     getFlavorByConfig(config) {
         return config ? this.getFlavorByName(config.name) : this.defaultFlavor;
     }
+
+    getFlavorByApplicationVersion(version) {
```
Comment (spectre007) [inline src/executable.js:45]: It returns an array of flavors, so we should name it `getFlavorsByApplicationVersion`

File: src/application.js [hunk 6]
```diff
@@ -96,9 +96,18 @@ export class Application extends NamedDefaultableInMemoryEntity {
...
         const tree = getAppTree(this.prop("name"));
-        return Object.keys(tree).map((key) => {
-            return new this.constructor.Executable({ ...tree[key], name: key });
-        });
+        return Object.keys(tree)
+            .filter((key) => {
+                const { supportedApplicationVersions } = tree[key];
+                return (
+                    !supportedApplicationVersions ||
+                    (supportedApplicationVersions &&
```
Comment (timurbazhirov) [inline src/application.js:104]: We can remove this line

File: src/executable.js [hunk 7]
```diff
@@ -41,4 +41,16 @@ export class Executable extends mix(NamedDefaultableInMemoryEntity).with(Runtime
     getFlavorByConfig(config) {
         return config ? this.getFlavorByName(config.name) : this.defaultFlavor;
     }
+
+    getFlavorsByApplicationVersion(version) {
+        const filteredFlavors = this.flavors.filter((flavor) => {
+            const supportedApplicationVersions = flavor.prop("supportedApplicationVersions");
+            return (
+                !supportedApplicationVersions ||
+                (supportedApplicationVersions && supportedApplicationVersions.includes(version))
```
Comment (timurbazhirov) [inline src/executable.js:50]: Same as above

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**